### PR TITLE
Feature/endpoint posts index

### PR DESCRIPTION
**Pull request by candelarialopezz**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -31,6 +31,9 @@
     mount_devise_token_auth_for 'User', at: 'auth'
     namespace :v1 do
       resources :users, only: [:index]
+      resources :users, only: [] do
+        resources :posts, only: [:index], controller: 'posts'
+      end
     end
```
Comment (daizuoheng) [inline config/routes.rb:37]:
```suggestion
      resources :users, only: [:index] do
        member do
          resources :posts, only: :index
        end
    end
```

File: spec/controllers/api/v1/posts_controller_spec.rb
[the remark this answers is not in the document]
Comment (EnzoCozza) [reply spec/controllers/api/v1/posts_controller_spec.rb:1]: And let's add a post that belongs to a different user, and check that it's not included in the response

**Review by daizuoheng — APPROVED** [review]

File: app/controllers/api/v1/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class PostsController < Api::V1::BaseController
+      def index
+        user = User.find(params[:user_id])
+        @posts = user.posts
+      rescue ActiveRecord::RecordNotFound
+        render json: { status: 404, message: 'User does not exist' }
```
Comment (EnzoCozza) [inline app/controllers/api/v1/posts_controller.rb:10]:
```suggestion
        render json: { message: 'User does not exist' }, status: :not_found
```

File: spec/controllers/api/v1/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    let!(:other_post) { create :post, user: other_user }
+    let!(:headers) { user.create_new_auth_token.merge('ACCEPT' => 'application/json') }
+
+    it 'with no token' do
+      get api_v1_user_posts_path(user.id)
+      expect(response).not_to have_http_status(:success)
+    end
+
+    it 'with invalid user id' do
+      get(api_v1_user_posts_path(32), headers:)
```
Comment (EnzoCozza) [inline spec/controllers/api/v1/posts_controller_spec.rb:20]: Let's use an ID that's guaranteed invalid (just in case). For example: 0 or -1

**Pull request merged**